editor/src/components/canvas/canvas-strategies/strategies/convert-to-absolute-and-move-strategy.tsx

```diff
@@ -613,3 +627,77 @@ function getFrameWithoutMargin(
   } as LocalPoint
   return offsetRect(frame, marginPoint)
 }
+
+function createSetParentsToFixedSizeCommands(
+  elementsToConvertToAbsolute: Array<ElementPath>,
+  metadata: ElementInstanceMetadataMap,
+): Array<CanvasCommand> {
+  if (elementsToConvertToAbsolute.length > 0) {
+    const firstJSXElementAncestors = stripNulls(
+      uniqBy(
+        elementsToConvertToAbsolute.map((path) => {
+          return EP.findAmongAncestorsOfPath(path, (p) => {
+            const element = MetadataUtils.getJSXElementFromMetadata(metadata, p)
+            if (element != null && isJSXElement(element)) {
+              return p
+            }
+            return null
+          })
+        }),
+        EP.pathsEqual,
+      ),
+    )
+
+    if (firstJSXElementAncestors.length == null) {
+      return []
+    }
+    return firstJSXElementAncestors.flatMap((ancestor) => {
+      const parentMetadata = MetadataUtils.findElementByElementPath(metadata, ancestor)
+      const parentElement = MetadataUtils.getJSXElementFromMetadata(metadata, ancestor)
+      const parentLocalFrame = parentMetadata?.localFrame
+      if (
+        parentElement != null &&
+        parentLocalFrame != null &&
+        isFiniteRectangle(parentLocalFrame)
+      ) {
+        const setWidthCommands = isCollapsingParent(parentElement, 'width')
+          ? [
+              setCssLengthProperty(
+                'always',
+                ancestor,
+                PP.create('style', 'width'),
+                setExplicitCssValue(cssPixelLength(parentLocalFrame.width)),
+                null,
+              ),
+            ]
+          : []
+        const setHeightCommands = isCollapsingParent(parentElement, 'width')
+          ? [
+              setCssLengthProperty(
+                'always',
+                ancestor,
+                PP.create('style', 'height'),
+                setExplicitCssValue(cssPixelLength(parentLocalFrame.height)),
+                null,
+              ),
+            ]
+          : []
+        return [...setWidthCommands, ...setHeightCommands]
+      }
+      return []
+    })
+  }
+
+  return []
+}
+
+const CollapsingWidthHeightValues = ['max-content', 'min-content', 'fit-content', 'auto']
+
+function isCollapsingParent(element: JSXElement, property: 'width' | 'height') {
+  const simpleAttribute = defaultEither(
+    null,
+    getSimpleAttributeAtPath(right(element.props), PP.create('style', property)),
```

Comment:
is it possible to base this on dom metadata? we could even extend specialSizeMeasurements if needed.